test/quantization/core/test_workflow_ops.py

```diff
@@ -840,26 +820,32 @@ def test_learnable_backward_per_channel_cuda(self, X):
         self._test_learnable_backward_per_channel(
             X_base, 'cuda', scale_base, zero_point_base, axis)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
-           X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
-           qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skip("temporarily disable the test")
-    def test_numerical_consistency_per_channel(self, device, X):
-        r"""Comparing numerical consistency between CPU quantize/dequantize op and the CPU fake quantize op
+    def test_numerical_consistency(self):
+        r"""Comparing numerical consistency between quantize/dequantize op and the fake quantize op across devices and dtypes
         """
-        np.random.seed(NP_RANDOM_SEED)
-        X, (scale, zero_point, axis, torch_type) = X
-        quant_min = torch.iinfo(torch_type).min
-        quant_max = torch.iinfo(torch_type).max
-
-        X = to_tensor(X, device)
-        scale = to_tensor(scale, device)
-        zero_point = torch.tensor(zero_point).to(dtype=torch.int64, device=device)
-        # quantize_linear and dequantize are only implemented in CPU
-        Y = torch.dequantize(torch.quantize_per_channel(X.cpu(), scale.cpu(), zero_point.cpu(), axis, torch_type))
-        Y_prime = torch.fake_quantize_per_channel_affine(
-            X, scale, zero_point, axis, quant_min, quant_max)
-        np.testing.assert_allclose(Y, Y_prime.cpu(), rtol=tolerance, atol=tolerance)
+        torch.random.manual_seed(NP_RANDOM_SEED)
+        torch_types = [torch.qint8, torch.quint8]
+        float_types = [torch.float, torch.float16, torch.float64]
+        zero_types = [torch.long]
+        devices = [torch.device('cpu'), torch.device('cuda')] if torch.cuda.is_available() else [torch.device('cpu')]
+        axis = 1
+        for i in range(20):
+            for torch_type, float_type, device, zero_type in itertools.product(torch_types, float_types, devices, zero_types):
+                X = torch.randn(3, 3, device=device).to(float_type)
+                scales = (10 * torch.randn(3, device=device)).abs()
+                scale = scales.mean().to(float).item()
+                zeros = (10 * torch.randn(3, device=device)).abs().to(dtype=zero_type)
+                zero = zeros.max().view(1).item()
+                quant_min = torch.iinfo(torch_type).min
+                quant_max = torch.iinfo(torch_type).max
+
+                Y = torch.dequantize(torch.quantize_per_tensor(X.to('cpu').to(torch.float), scale, zero, torch_type)).to(device).to(float_type)
+                Y_prime = torch.fake_quantize_per_tensor_affine(X, scale, zero, quant_min, quant_max)
+                self.assertEqual(Y, Y_prime, "Difference found between dequant+quant_per_tensor and fake_quantize_per_tensor")
+
+                Y = torch.dequantize(torch.quantize_per_channel(X.to('cpu').to(torch.float), scales.to('cpu'), zeros.to('cpu'), axis, torch_type)).to(device).to(float_type)
```

Comment:
optional nit: would be nice to have per-tensor and per-channel in individual test cases instead of together, to keep tests single purposed